mypy/nodes.py

```diff
@@ -2099,25 +2094,9 @@ def get_containing_type_info(self, name: str) -> 'Optional[TypeInfo]':
                 return cls
         return None
 
-    def record_subtype_cache_entry(self, left: 'mypy.types.Instance',
-                                   right: 'mypy.types.Instance',
-                                   proper_subtype: bool = False) -> None:
-        if proper_subtype:
-            self._cache_proper.add((left, right))
-        else:
-            self._cache.add((left, right))
-
-    def is_cached_subtype_check(self, left: 'mypy.types.Instance',
-                                right: 'mypy.types.Instance',
-                                proper_subtype: bool = False) -> bool:
-        if not proper_subtype:
-            return (left, right) in self._cache
-        return (left, right) in self._cache_proper
-
     def reset_subtype_cache(self) -> None:
         for item in self.mro:
-            item._cache = set()
-            item._cache_proper = set()
+            TypeState.reset_caches(item)
```

Comment:
We could simplify module dependencies a bit by moving this method to `TypeState`. This would have the benefit of `TypeInfo` not needing any knowledge of `TypeState` (together with the above idea). Conceptually then `TypeState` would only hold information about speeding up certain `TypeInfo` operations, but it would be separate from `TypeInfo`. It's better if TypeInfo doesn't need to be aware of operations available for TypeInfos.

In the future we can move additional 'transient' global state from TypeInfos to TypeState (or some other place), such as `inferring` and `assuming` (it's best to not do it in this PR to avoid making this big).